### PR TITLE
feat(group): Add feature to fetch user ID from user name

### LINKED ISSUE #59: Managing members by member email or username
Is it possibly to use useremail or username instead of userid?
```
kind: Member
metadata:
  name: admin-member
spec:
  forProvider:
    groupIdRef:
      name: crossplane-example
    userID: example@mail.ru
    accessLevel: 50
  providerConfigRef:
    name: gitlab-provider
  writeConnectionSecretToRef:
    name: gitlab-group-example-member
    namespace: crossplane-system
```

### PR DESCRIPTION
**Pull request by IntaSlapna**
### Description of your changes

Issue #59 - part 2: added option for creating Group Member by UserName.

Needed label "braking-change" as well, because MemberParameters.UserID changed from int to *int. 

-->
Fixes #59 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Unit-tests used. `make reviewable test` PASS. 

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by MisterMX — CHANGES_REQUESTED** [review]

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,6 @@ NPROCS ?= 1
 # parallel can lead to high CPU utilization. by default we reduce the parallelism
 # to half the number of CPU cores.
 GO_TEST_PARALLEL := $(shell echo $$(( $(NPROCS) / 2 )))
-
```
[the remark this answers is not in the document]
Comment (IntaSlapna) [reply Makefile:35]: I suppose there was GO_REQUIRED_VERSION := 1.20 
Which we don't push, right?

File: apis/projects/v1alpha1/member_type.go [hunk 2]
```diff
@@ -40,8 +40,12 @@ type MemberParameters struct {
 	ProjectIDSelector *xpv1.Selector `json:"projectIdSelector,omitempty"`
 
 	// The user ID of the member.
-	// +immutable
-	UserID int `json:"userID"`
+	// +optional
+	UserID *int `json:"userId,omitempty"`
```
[the remark this answers is not in the document]
Comment (IntaSlapna) [reply apis/projects/v1alpha1/member_type.go:44]: changed

File: pkg/clients/users/user.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	if err != nil {
+		return nil, errors.Wrap(err, errFetchFailed)
+	}
+	if len(userArr) == 0 || len(userArr) > 1 {
+		return nil, errors.Errorf(errPullUserID, len(userArr))
+	}
+
+	pulledUserID := userArr[0].ID
+	if pulledUserID == 0 {
+		return nil, errors.New(errFetchFailed)
```
[the remark this answers is not in the document]
Comment (MisterMX) [reply pkg/clients/users/user.go:55]: But if we don't receive the correct user ID shouldn't `len(userArr) == 0`? And if there is any element in the array then it means success?

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (MisterMX) [reply Makefile:35]: No, as the diff shows, there was just an empty line which got removed. But that is not a necessary change.

File: pkg/clients/users/user.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (IntaSlapna) [reply pkg/clients/users/user.go:55]: It means that it is not needed double check? 
I will take out this "if" then.

[the remark this answers is not in the document]
Comment (IntaSlapna) [reply pkg/clients/users/user.go:55]: And yes, the idea was if there is only one element in array it means that we successfully have matched user by username and can extract usedID.

[the remark this answers is not in the document]
Comment (MisterMX) [reply pkg/clients/users/user.go:55]: Sounds good

**Review by MisterMX — APPROVED** [review]
LGTM. Thank you very much for your contribution @IntaSlapna!

**Pull request merged**